monai/networks/nets/finetune_fc.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple, Union
+
+import torch
+
+
+class FinetuneFC(torch.nn.Module):
+    """
+    Wrapper to customize the fully connected layer of 2D classification model or replace it by convolutional layer.
+
+    Args:
+        model: PyTorch model with fully connected layer at the end, typically, it can be a pretrained model in
+            Torchvision, like: ``resnet18``, ``resnet34m``, ``resnet50``, ``resnet101``, ``resnet152``, etc.
+            model details: https://pytorch.org/vision/stable/models.html.
+        n_classes: number of classes for the last classification layer. Default to 1.
+        use_conv: whether use convolutional layer to replace the FC layer, default to False.
+        pool_size: if using convolutional layer to replace the FC layer, it defines the kernel size for `AvgPool2d`
+            to replace `AdaptiveAvgPool2d`. Default to (7, 7).
+        pool_stride: if using convolutional layer to replace the FC layer, it defines the stride for `AvgPool2d`
+            to replace `AdaptiveAvgPool2d`. Default to 1.
+
+    """
+
+    def __init__(
+        self,
+        model: torch.nn.Module,
+        n_classes: int = 1,
+        use_conv: bool = False,
+        pool_size: Union[int, Tuple[int, int]] = (7, 7),
+        pool_stride: Union[int, Tuple[int, int]] = 1,
+    ):
+        super().__init__()
+        layers = list(model.children())
+
+        # check if the model is compatible
+        if not str(layers[-1]).startswith("Linear"):
+            raise ValueError("input model does not have a Linear layer at the end.")
```

Comment:
perhaps this check is not needed, the final layers might be some activation/norm layers

Reply:
if FC is not the last layer, how do we detect the FC layer?

Thanks.

Reply:
the wrapper is to remove the existing final layers, and construct new ones, as long as `self.features = torch.nn.Sequential(*layers[:-2])` produces some feature maps in shape BxCxSpatialdims it should be ok.

Reply:
Hi @wyli ,

But if the last several layers are uncertain, how can we make sure the feature maps is [:**-2**]?
For example, if has activation layer, I think we should use [:-3]?
And should we also add the activation layer after the modified FC layer if original model has it?
I feel maybe it's not easy to support the case that FC is not last layer, or maybe I didn't get your point clearly?

Thanks.

Reply:
this is a low-level component, this class could be configured to do two things: (1) replace the last two layers with a 'pooling + conv'; or (2) replace the last layer with a 'linear'. whether the input `model` is suitable for this component should be decided by the upper-level components and probably depending on the applications... we can add an optional ADN layer at the end but it could also be a postprocessing step, so it's fine to omit that for now.

Reply:
Sure, I will try to update it ASAP today.
Thanks for your suggestions.

Reply:
added support for 3D and more generic cases in the latest commit.
Thanks.